lib/singletonable.js

```diff
@@ -0,0 +1,25 @@
+// Add lazy singletons to Constructors. This adds a makeSingleton class
+// method, which will create a singleton property of the given name, passing
+// the remaining arguments to the constructor.
+//
+// The singleton is constructed on first access, through
+// Object.defineProperty. This defers initialization, simplifying dependency
+// management.
+
+module.exports = function (Constructor) {
+  Constructor.makeSingleton = function (name, varargs) {
+    // Singleton should be of whatever makeSingleton was called on, not
+    // necessarily what was passed to singletonable()
+    var Child = this;
+    varargs = arguments._.tail();
+
+    // Storage for the the singleton object, once it is created
+    var singleton;
```

Comment:
In a previous iteration, I had set configurable: true, thinking that would make it so subclasses could override it. But then I realized that that's not necessary for the subclass case, and took it off. Mocking is a good point though, so I think you're right.

I have a trick for this kind of situation where you define a property as a configurable property, but then set it when it's accessed, so it acts exactly like normal property. That'll be good because then we can use autoMock too.